src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -841,6 +863,30 @@ public void testCacheOnViolationSuppression() throws Exception {
         }
     }
 
+    @Test
+    public void testHaltOnException() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(CheckWhichThrowsError.class);
+
+        final DefaultConfiguration treeWalkerConfig = createModuleConfig(TreeWalker.class);
+        treeWalkerConfig.addChild(checkConfig);
```

Comment:
Can we replace these 2 lines with `createTreeWalkerConfig`?

Reply:
Moreover, since the default value of `haltOnException` is `true` both `treeWalkerConfig` and `checkerConfig` are unnecessary.